src/frontend/with-page-auth-required.tsx

```diff
@@ -66,18 +81,26 @@ export type WithPageAuthRequired = <P extends object>(
 const withPageAuthRequired: WithPageAuthRequired = (Component, options = {}) => {
   return function withPageAuthRequired(props): JSX.Element {
     const router = useRouter();
-    const { returnTo = router.asPath, onRedirecting = defaultOnRedirecting, loginUrl = '/api/auth/login' } = options;
-    const { user, isLoading } = useUser();
+    const {
+      returnTo = router.asPath,
+      onRedirecting = defaultOnRedirecting,
+      onError = defaultOnError,
+      loginUrl = '/api/auth/login'
+    } = options;
+    const { user, error, isLoading } = useUser();
 
     useEffect(() => {
-      if (user || isLoading) return;
+      if ((user && !error) || isLoading) return;
 
       (async (): Promise<void> => {
         await router.push(`${loginUrl}?returnTo=${returnTo}`);
       })();
-    }, [user, isLoading, router, loginUrl, returnTo]);
+    }, [user, error, isLoading]);
+
+    if (error) return onError(error);
+    else if (user) return <Component {...props} />;
```

Comment:
this can be just an `if`

Reply:
Changed in 6ceda646787bd2c0676ecc0a288522b2645e49de.